src/pages/room/[roomId]/[threadId].tsx

```diff
@@ -1,27 +1,88 @@
 import Head from "next/head";
-import { Room, PrismaClient, RoomUser } from "@prisma/client";
+import {
+  Room,
+  RoomUser,
+  PrismaClient,
+  User,
+  Thread,
+  Message,
+} from "@prisma/client";
 import { GetServerSideProps, NextPage } from "next";
-import Layout from "../../../components/layout";
+import dynamic from "next/dynamic";
+import Button from "react-bootstrap/Button";
+
+const RoomPage = dynamic(() => import("../[roomId]"));
 
 interface ThreadProps {
-  room: Room & { users: RoomUser[] };
-  thread: Number;
+  room: Room & {
+    users: (RoomUser & { user: User })[];
+    threads: (Thread & { creator: User })[];
+  };
+  thread: Thread;
+  /*you can use the Omit utility type to remove the user property from the Message type, and then add it back with the correct type */
+  messages: (Omit<Message, "user"> & { user: User })[];
 }
 
-const Thread: NextPage<ThreadProps> = ({ room, thread }) => {
-  return (
-    <Layout>
-      <Head>
-        <title>{`TUC Forum - ${room.name} - ${thread}`}</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+function formatTime(date: Date) {
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+  return hours + ":" + minutes;
+}
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#instance_methods
Es gibt ganz viele `to[...]String()` Methoden, z.B.:
```js
toLocaleString('de', { hour:'numeric', minute:'numeric' }) // -> '21:11'
toLocaleString('en', { hour:'numeric', minute:'numeric' }) // -> '9:11 PM'
```

oder noch besser: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString